pkg/app/server.go

```diff
@@ -99,9 +101,25 @@ func RunKubeStateMetrics(ctx context.Context, opts *options.Options, factories .
 			ConstLabels: prometheus.Labels{"handler": "metrics"},
 		}, []string{"method"},
 	)
+	configHash := promauto.With(ksmMetricsRegistry).NewGauge(
+		prometheus.GaugeOpts{
+			Name: "kube_state_metrics_config_hash",
+			Help: "Hash of the currently loaded configuration.",
+		})
+	configSuccess := promauto.With(ksmMetricsRegistry).NewGauge(
+		prometheus.GaugeOpts{
+			Name: "kube_state_metrics_config_last_reload_successful",
+			Help: "Whether the last configuration reload attempt was successful.",
+		})
+	configSuccessTime := promauto.With(ksmMetricsRegistry).NewGauge(
+		prometheus.GaugeOpts{
+			Name: "kube_state_metrics_config_last_reload_success_timestamp_seconds",
```

Comment:
How about `kube_state_metrics_last_config_uptime`?
So that we can easily know whether there are some changes to config recently.

It equals to current_timestamp - kube_state_metrics_config_last_reload_success_timestamp_seconds

Reply:
I prefer to keep it as it, as a timestamp over a counter is usually preferred and has a couple of advantages:
- few static values inside the TSDB for this metric
- Resets can also be detected if the timestamp changes
- You never have the same counter twice.
See also: https://prometheus.io/docs/practices/instrumentation/#timestamps-not-time-since